src/shogun/lib/observers/ObservedValue.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Giovanni De Toni
+ */
+
+#ifndef SHOGUN_OBSERVEDVALUE_H
+#define SHOGUN_OBSERVEDVALUE_H
+
+#include <shogun/base/SGObject.h>
+
+namespace shogun {
+
+	template<class T>
+	class ObservedValueTemplated;
+
+	/**
+ 	* Observed value which is emitted by algorithms.
+ 	*/
+	class ObservedValue : public CSGObject {
```

Comment:
cool, after this PR is in, @theartful can try to split the params as a mixin and we can try to remove the base class here